### ISSUE TITLE
How do unknown nodes obtain graph embeddings from a known network?

### ISSUE THREAD

**Issue by zhuzn** [Zhaolab-GitHub/ZHMolGraph#1]
Hello,

Thanks for the nice software.

There are some protein and RNA sequences that do not exist in the known network. How does the model obtain their graph embeddings from the pre-trained network?

**Comment by clumzysoap** [comment]
I meet this problem weeks ago.After reading the code ,I think the unknown protein and RNA sequences are seen as isolated node in the graph

**Comment by Zhaolab-GitHub** [comment]
When constructing the unknown test set TheNovel, we ensured that the proteins and RNAs present in the training graph were entirely removed. During testing on this unseen dataset, the unknown proteins and RNAs are isolated as “orphan”&nbsp;nodes in the training graph, with no known interactions.&nbsp;


 -------------------------------------------
Yunjie Zhao, Ph.D.
Professor
Institute of Biophysics
Department of Physics
Central China Normal University
 

&nbsp;
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: ***@***.***&gt;;
Date: &nbsp;Tue, Apr 8, 2025 00:44 AM
To: ***@***.***&gt;; 
Cc: ***@***.***&gt;; 
Subject: &nbsp;Re: [Zhaolab-GitHub/ZHMolGraph] How do unknown nodes obtain graph embeddings from a known network? (Issue #1)

&nbsp;

   
I meet this problem weeks ago.After reading the code ,I think the unknown protein and RNA sequences are seen as isolated node in the graph

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***&gt;
  clumzysoap left a comment (Zhaolab-GitHub/ZHMolGraph#1)
 
I meet this problem weeks ago.After reading the code ,I think the unknown protein and RNA sequences are seen as isolated node in the graph

**Comment by Zhaolab-GitHub** [comment]
We used the SAGE layer as the network module to obtain the graph embeddings.
SAGE is a general inductive framework that leverages node feature information to efficiently generate node embeddings for previously unseen data. Therefore, their graph embeddings can also be derived through the trained SAGE layer for protein and RNA nodes that are not present in the known interaction network.




 -------------------------------------------
Yunjie Zhao, Ph.D.
Professor
Institute of Biophysics
Department of Physics
Central China Normal University
 

&nbsp;
&nbsp;
&nbsp;
------------------&nbsp;Original&nbsp;------------------
From: ***@***.***&gt;;
Date: &nbsp;Fri, Mar 14, 2025 05:27 PM
To: ***@***.***&gt;; 
Cc: ***@***.***&gt;; 
Subject: &nbsp;[Zhaolab-GitHub/ZHMolGraph] How do unknown nodes obtain graph embeddings from a known network? (Issue #1)

&nbsp;

   
Hello,
 
Thanks for the nice software.
 
There are some protein and RNA sequences that do not exist in the known network. How does the model obtain their graph embeddings from the pre-trained network?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***&gt;
  zhuzn created an issue (Zhaolab-GitHub/ZHMolGraph#1)
 
Hello,
 
Thanks for the nice software.
 
There are some protein and RNA sequences that do not exist in the known network. How does the model obtain their graph embeddings from the pre-trained network?

**Comment by zhuzn** [comment]
> We used the SAGE layer as the network module to obtain the graph embeddings.
> SAGE is a general inductive framework that leverages node feature information to efficiently generate node embeddings for previously unseen data. Therefore, their graph embeddings can also be derived through the trained SAGE layer for protein and RNA nodes that are not present in the known interaction network.
> […](#)
> 
> -------------------------------------------
> Yunjie Zhao, Ph.D.
> Professor
> Institute of Biophysics
> Department of Physics
> Central China Normal University
> 
> 
> &nbsp;
> &nbsp;
> &nbsp;
> ------------------&nbsp;Original&nbsp;------------------
> From: ***@***.***&gt;;
> Date: &nbsp;Fri, Mar 14, 2025 05:27 PM
> To: ***@***.***&gt;;
> Cc: ***@***.***&gt;;
> Subject: &nbsp;[Zhaolab-GitHub/ZHMolGraph] How do unknown nodes obtain graph embeddings from a known network? (Issue [#1](https://github.com/Zhaolab-GitHub/ZHMolGraph/issues/1))
> 
> &nbsp;
> 
> 
> Hello,
> 
> Thanks for the nice software.
> 
> There are some protein and RNA sequences that do not exist in the known network. How does the model obtain their graph embeddings from the pre-trained network?
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.Message ID: ***@***.***&gt;
>   zhuzn created an issue ([Zhaolab-GitHub/ZHMolGraph#1](https://github.com/Zhaolab-GitHub/ZHMolGraph/issues/1))
> 
> Hello,
> 
> Thanks for the nice software.
> 
> There are some protein and RNA sequences that do not exist in the known network. How does the model obtain their graph embeddings from the pre-trained network?
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you are subscribed to this thread.Message ID: ***@***.***&gt;

Many thanks for the replies.
